### PR TITLE
docs: Declarative argo clusters docs update for EKS

### PR DESCRIPTION
**Pull request by fideloper**
I had a hard time getting the declarative setup to work when adding EKS clusters  to Argo CD. I updated the docs to clarify a few things:

1. Use Access Entries instead of depreciated `aws-auth` ConfigMap
2. Made the requirements a bit of a punch list with corresponding descriptions on each section
3. Updated Argo CD's role's example trust policy to show it needed to be able to assume itself
4. Wording updates/changes to be more specific about what examples relate to which role (etc)
5. Added a third Service Account which I believe needs to get an AWS role added to its annotation.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: docs/operator-manual/declarative-setup.md [hunk 1]
```diff
@@ -669,27 +699,14 @@ for it.
...
-  }
-```
+The 3 service accounts need to be modified to include an annotation with the Argo CD management role ARN.
 
-Example service account configs for `argocd-application-controller` and `argocd-server`.
+Here's an example service account configurations for `argocd-application-controller`, `argocd-applicationset-controller`, and `argocd-server`.
 
 !!! warning
-    Once the annotations have been set on the service accounts, both the application controller and server pods need to be restarted.
+Once the annotations have been set on the service accounts, the application controllers and server pods need to be restarted.
```
Comment (nitishfy) [inline docs/operator-manual/declarative-setup.md:709]:
```suggestion
Once the annotations have been set on the service accounts, the application controller and server pods need to be restarted.
```

It should be application controller.

**Review by nitishfy — CHANGES_REQUESTED** [review]
Please sign-off the commits too.

File: docs/operator-manual/declarative-setup.md [hunk 2]
```diff
@@ -625,8 +625,8 @@ metadata:
     argocd.argoproj.io/secret-type: cluster
 type: Opaque
 stringData:
-  name: "mycluster.example.com"
-  server: "https://mycluster.example.com"
+  name: "eks-cluster-name-for-argo"
+  server: "https://xxxyyyzzz.xyz.some-region.eks.amazonaws.com"
```
[the remark this answers is not in the document]
Comment (todaywasawesome) [reply docs/operator-manual/declarative-setup.md:629]: I believe it is yes!

File: docs/operator-manual/declarative-setup.md (hunk 1 above)
Comment (todaywasawesome) [reply docs/operator-manual/declarative-setup.md:709]: +1

**Review by todaywasawesome — CHANGES_REQUESTED** [review]
Great addition, just a few nits to get this merged.

File: docs/operator-manual/declarative-setup.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (fideloper) [reply docs/operator-manual/declarative-setup.md:629]: The name is consistent across the EKS docs now - this actually addresses some confusion I had while setting it up so dummies like me I think make it useful.

(Same with making the `server` endpoint so obviously an AWS EKS endpoint).

**Comment by fideloper** [comment]
Thanks for you input and help! I believe I've captured/incorporated the feedback.

I've resigned commits via `git rebase --signoff HEAD~3`, so the commits should be handled there.

**Review by todaywasawesome — APPROVED** [review]
Thanks for the quick turnaround. I'm good, @nitishfy ?

**Review by nitishfy — APPROVED** [review]
LGTM, thanks!

**Comment by namnt2307** [comment]
I'm confused about this part `Updated Argo CD's role's example trust policy to show it needed to be able to assume itself
`  when checking the documentation. Can you explain why do Argo need to be able to assume itself?

**Comment by fideloper** [comment]
@namnt2307 well, you see, when a mommy and daddy role love each other very much, they decide to make new roles. but sometimes a daddy role can't find a mommy role and has to do the job himself.

(Honestly I don't know why it's assuming itself, it could be an artifact of the code there, where a role is re-used to control the cluster argo is running on, and thus attempts to assume itself. AWS being AWS of course won't let you do that unless you let explicitly set it in its IAM permissions).

I don't personally have time to dig into that but I hope someone does 😂. IIRC, I was just following error messages I saw in ArgoCD.

**Pull request merged**